app/src/backend/environment/hooks/useEnvironmentManager.ts

```diff
@@ -0,0 +1,129 @@
+  const allEnvironmentData = useSelector(getAllEnvironmentData);
+
+  const ownerId = useMemo(
+    () => (currentlyActiveWorkspace.id ? `team-${currentlyActiveWorkspace.id}` : user?.details?.profile?.uid),
+    [currentlyActiveWorkspace.id, user?.details?.profile?.uid]
+  );
+
+  useEffect(() => {
+    unsubscribeListener?.();
+    unsubscribeListener = attatchEnvironmentVariableListener(ownerId, (environmentMap) => {
+      dispatch(environmentVariablesActions.setAllEnvironmentData({ environmentMap }));
+    });
+
+    return () => {
+      unsubscribeListener();
+    };
+  }, [dispatch, ownerId]);
+
+  useEffect(() => {
+    if (!user.loggedIn) {
+      unsubscribeListener?.();
+      dispatch(environmentVariablesActions.resetState());
+    }
+  }, [dispatch, user.loggedIn]);
+
+  const getCurrentEnvironment = () => {
+    return {
+      currentEnvironmentName,
+      currentEnvironmentId,
+    };
+  };
+
+  const addNewEnvironment = async (newEnvironment: string) => {
+    return setEnvironmentInDB(ownerId, newEnvironment).catch((err) => {
+      console.error("Error while setting environment in db", err);
+    });
+  };
+
+  const setCurrentEnvironment = (newEnvironment: string) => {
+    dispatch(environmentVariablesActions.setEnvironment({ environmentName: newEnvironment }));
+  };
+
+  const setVariables = async (
+    environmentId: string,
+    variables: Record<string, Omit<EnvironmentVariableValue, "type">>
+  ) => {
+    const newVariables: EnvironmentVariables = Object.fromEntries(
+      Object.entries(variables).map(([key, value]) => {
+        return [key, { localValue: value.localValue, syncValue: value.syncValue, type: typeof value.syncValue }];
+      })
+    );
+
+    return setEnvironmentVariablesInDB(ownerId, {
+      newVariables,
+      environmentId,
+    })
+      .then(() => {
+        dispatch(
+          environmentVariablesActions.setVariablesInEnvironment({
+            newVariables,
+            environmentId,
+          })
+        );
+      })
+      .catch((err) => {
+        toast.error("Error while setting environment variables.");
+        Logger.error("Error while setting environment variables in db", err);
+        console.error("Error while setting environment variables in db", err);
+      });
+  };
+
+  const removeVariable = async (environmentId: string, key: string) => {
+    return removeEnvironmentVariableFromDB(ownerId, { environmentId, key })
+      .then(() => {
+        dispatch(environmentVariablesActions.removeVariableFromEnvironment({ key, environmentId }));
+      })
+      .catch((err) => {
+        toast.error("Error while removing environment variables.");
+        Logger.error("Error while removing environment variables from db", err);
+        console.error("Error while removing environment variables from db", err);
+      });
+  };
+
+  const renderString = <T>(template: string | Record<string, any>): T => {
```

Comment:
Rename this to `renderWithVars()` or `render` only as this is rendering not only string but object types too